docs/design/nab_and_nar_status_update.md

```diff
@@ -0,0 +1,192 @@
+It is always a one well defined value, that is intended to be a comprehensive state of a NAB or NAR object.
+
+Those are are the possible values for phase:
+
+| **Value** | **Description** |
+|-----------|-----------------|
+| New | *NonAdminBackup/NonAdminRestore* resource was accepted by the NAB/NAR Controller, but it has not yet been validated by the NAB/NAR Controller |
+| BackingOff | *NonAdminBackup/NonAdminRestore* resource was invalidated by the NAB/NAR Controller, due to invalid Spec. NAB/NAR Controller will not reconcile the object further, until user updates it |
+| Created | *NonAdminBackup/NonAdminRestore* resource was validated by the NAB/NAR Controller and Velero *Backup/restore* was created. The Phase will not have additional information about the *Backup/Restore* run |
+
+### Conditions
+
+The `conditions` field is a list of conditions.
+One NAB/NAR object may have multiple conditions.
+It is more granular knowledge of the NAB/NAR object and represents the array of the conditions through which the object has or has not passed.
+
+Each `condition` data is composed by the following:
+
+| **Field name** | **Description** |
+|----------------|-----------------|
+| type | The `NonAdminCondition` of the condition |
+| status | represents the state of individual condition. One of `True`, `False` or `Unknown`. |
+| lastTransitionTime | Timestamp for when this condition was created/updated. |
+| reason | Machine-readable, UpperCamelCase text indicating details about the last status transition. |
+| message | Human-readable message indicating details about the last status transition. |
+
+Those are are the possible values for `NonAdminCondition`:
+
+| **Value** | **Description** |
+|-----------|-----------------|
+| Accepted | The NonAdminBackup/NonAdminRestore object was accepted by the controller, but the Velero Backup/Restore may have not yet been created |
+| Queued | The Velero Backup/Restore was created successfully. At this stage errors may still occur either from the Velero not accepting object or during backup/restore procedure. |
+
+### Velero object reference
+
+NonAdminBackup/NonAdminRestore `status` contains reference to the related Velero Backup/Restore.
+
+NonAdminBackup `status.veleroBackup` contains `name`, `namespace` and `status`.
+- `status.veleroBackup.name` represents the name of the `VeleroBackup` object.
+- `status.veleroBackup.namespace` represents the namespace in which the `VeleroBackup` object was created.
+- `status.veleroBackup.status` field is a copy of the `VeleroBackup` object status.
+
+The format of those fields allows to interact with that Backup using `oc` or `velero` commands as follows:
+
+```yaml
+status:
+  veleroBackup:
+    name: nab-nacproject-c3499c2729730a
+    namespace: openshift-adp
+    status:
+     ...
+```
+
+```shell
+velero backup describe -n openshift-adp nab-nacproject-c3499c2729730a
+```
+
+Similarly, NonAdminRestore `status.veleroRestore` contains `name`, `namespace` and `status`.
```

Comment:
I don't see here how the NAR is related to NAB.

Reply:
here, we are just telling that both NAB and NAR have Velero reference in their status, not related. Is it not worded correctly?

Reply:
Sorry for not being clear:
 1. The user creates `NonAdminBackup` let'c call it `a-uuid`, our controller creates Velero `Backup`. Both objects either in Label and/or Status, contains references with the value `a-uuid`. At this time we don't really care much about Velero `Backup` name. It should and probably will be `a-uuid`, but it's not 100% guaranteed.

2. The user would like to restore Backup. The pre-requisite is that `NonAdminBackup` object exists in the user's namespace. The existence of the `NonAdminBackup` object is guaranteed by either simply being there from the time where user created it or from the time Sync Controller created it for the user. In this situation `NonAdminRestore` object should reference `NOT` the Velero `Backup` but instead the `NonAdminBackup`. That is why I think it's incorrect to tell that Similarly it's the same as NonAdminBackup.

To rephrase NonAdminBackup should reference Velero Backup and NonAdminRestore should reference NonAdminBackup